### PR TITLE
Style and refactor home screen

### PR DESCRIPTION
**Pull request by david-qing**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :cloud: :cloud: Description :cloud: :cloud:
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Refactored the home screen such that it looks like the Figma. 
- Removed Views used for styling in CategoryCard, and replaced them with positioning in the CSS. 
- Implemented a color prop, such that the background color of each CategoryCard can be customized.
- Simplified the HomeScreen code with arrays of styling. 

### Screenshots :calling:
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @leexesther at the bottom."

Screen Shot 2023-03-10 at 3 44 35 AM

## :zap: How to review :zap:
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Make sure that this app works on any device you can test it on. Do so using `npx expo start` to start the application, and test for visual bugs. 

## :seedling: Next steps :seedling:
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Decide the onPress functionality for each of the category cards; where should each go? A search screen with the filter set to the category? This is a design decision: @leexesther

## Relevant Links :link:

N/A

### Online sources :information_source:
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

N/A

### Related PRs :raised_hands:
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

https://github.com/calblueprint/cultural-survival-mobile/pull/3

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @davidqing6432
@leexesther

### CODE REVIEW

File: src/components/CategoryCard.tsx [hunk 1]
```diff
@@ -1,56 +1,53 @@
...
   StyleSheet,
   ImageSourcePropType,
   Pressable,
 } from 'react-native';
 
 const styles = StyleSheet.create({
-  categoryContainer: {
-    display: 'flex',
+  container: {
+    position: 'relative',
```
Comment (wangannie) [inline src/components/CategoryCard.tsx]: Is this supposed to be relative?

File: src/components/CategoryCard.tsx [hunk 2]
```diff
@@ -1,56 +1,53 @@
...
+    marginRight: -24,
+    marginBottom: -24,
   },
 });
 
 type CategoryProps = {
   name: string;
   pressFunction: (event: GestureResponderEvent) => void;
   img: ImageSourcePropType;
+  color: string;
```
Comment (wangannie) [inline src/components/CategoryCard.tsx]: Nit: `backgroundColor` for clarity

File: src/components/CategoryCard.tsx [hunk 3]
```diff
@@ -1,56 +1,53 @@
...
+  image: {
+    position: 'absolute',
+    bottom: 0,
+    right: 0,
+    marginRight: -24,
+    marginBottom: -24,
   },
 });
 
 type CategoryProps = {
```
Comment (wangannie) [inline src/components/CategoryCard.tsx]: Since you're refactoring this anyways, I would change this to `CategoryCardProps` so it stays consistent with the convention of `{FullComponentName}Props`

File: src/components/CategoryCard.tsx [hunk 4]
```diff
@@ -1,56 +1,53 @@
...
+    borderRadius: 8,
+    height: 120,
+    width: 166,
+    padding: 14,
+    margin: 8,
   },
-  imgStyle: {
-    width: '80%',
-    height: '80%',
+  text: {
```
Comment (wangannie) [inline src/components/CategoryCard.tsx]: Nit: I would rename this to `label`, `title` or something more descriptive to how the text functions in the card. Text could be anything.

File: src/screens/HomeScreen/Home.tsx [hunk 5]
```diff
@@ -15,43 +15,50 @@ function HomeScreen({ navigation }: RootStackScreenProps<'Home'>) {
...
           <CategoryCard
             name="Language & Culture"
             pressFunction={() => navigation.navigate('Home')}
             img={languageImage}
+            color="#FBBE47"
           />
           <CategoryCard
             name="Education"
-            pressFunction={() => navigation.navigate('Home')}
+            pressFunction={() => navigation.navigate('Play')}
```
Comment (wangannie) [inline src/screens/HomeScreen/Home.tsx:47]: Was this navigation supposed to change?

File: src/screens/HomeScreen/Home.tsx [hunk 6]
```diff
@@ -15,43 +15,50 @@ function HomeScreen({ navigation }: RootStackScreenProps<'Home'>) {
...
             img={educationImage}
+            color="#74944A"
           />
           <CategoryCard
             name="Land Rights"
-            pressFunction={() => navigation.navigate('Home')}
+            pressFunction={() => navigation.navigate('Welcome')}
```
Comment (wangannie) [inline src/screens/HomeScreen/Home.tsx:53]: Was this navigation supposed to change?

File: src/screens/HomeScreen/Home.tsx [hunk 7]
```diff
@@ -15,43 +15,50 @@ function HomeScreen({ navigation }: RootStackScreenProps<'Home'>) {
...
+        <Text style={[globalStyles.h2, styles.heading]}>
+          Welcome to Cultural Survival.
         </Text>
-        <Text style={styles.padding2}>
-          <Text style={globalStyles.h4}>Categories</Text>
+        <Text style={[globalStyles.h4, styles.subheading]}>
+          Browse all categories
         </Text>
-
+        {/* TODO: Change color from a string to importing color from global styles */}
```
Comment (wangannie) [inline src/screens/HomeScreen/Home.tsx:24]: YAY 👍🏻 helpful TODO comment

File: src/screens/HomeScreen/styles.tsx [hunk 8]
```diff
@@ -7,15 +7,12 @@ export default StyleSheet.create({
...
     justifyContent: 'space-evenly',
-    paddingTop: '5%',
+    paddingTop: 16,
   },
-  padding: {
-    padding: '2%',
-    alignSelf: 'center',
-  },
-  padding2: {
+  heading: { paddingLeft: 16, paddingTop: 16 },
```
Comment (wangannie) [inline src/screens/HomeScreen/styles.tsx]: Nit: I would separate this into new lines to be consistent with everything else.

File: src/screens/HomeScreen/styles.tsx [hunk 9]
```diff
@@ -7,15 +7,12 @@ export default StyleSheet.create({
...
+  subheading: {
+    paddingLeft: 16,
     marginTop: '2%',
-    paddingLeft: '10%',
     textAlign: 'left',
   },
   outerContainer: {
```
Comment (wangannie) [inline src/screens/HomeScreen/styles.tsx]: It's unclear what `container` vs `outerContainer` are. I would at least rename `container` to `cardContainer` or something more specific like that. 

As for the `outerContainer`, in the past, I've used a convention where the outer-most container is always called `main` -- I would recommend using that here! Or you could just rename this to `container`. Happy to discuss!

File: src/components/CategoryCard.tsx [hunk 10]
```diff
@@ -1,56 +1,53 @@
...
-          <Image style={styles.imgStyle} source={img} />
-        </View>
-        <Text style={styles.txtBox}>{name}</Text>
-      </View>
+    <Pressable
+      style={[styles.container, { backgroundColor: color }]}
+      onPress={pressFunction}
+    >
+      <Text style={styles.text}>{name}</Text>
+      <Image style={styles.image} source={img} />
```
Comment (wangannie) [inline src/components/CategoryCard.tsx]: Right now, I'm seeing the image cover the text. Adding `zIndex: 1` to the text should fix it. ([Expo docs](https://docs.expo.dev/ui-programming/z-index/))


IMG_C7724C6E29C5-1

**Review by wangannie — APPROVED** [review]
This looks awesome! My comments are mostly nits and should _hopefully_ be quick to address. Great work : )
